### ISSUE TITLE
Editable GPS Field Causing issues

### ISSUE THREAD

**Issue by irshadshalu** [IEEEKeralaSection/rescuekerala#225]
### Subject of the issue
As warned, making GPS field as editable is causing havoc as many users are entering random URLs/addresses to the location. Refer the screenshot for more information
![screenshot_2018-08-17-00-30-58-204_com android chrome](https://user-images.githubusercontent.com/8298478/44229408-9b369c00-a1b5-11e8-9e6e-cbbd75a08186.png)
This is caused by  #160 

### Steps to reproduce
Open keralarescue.in/requests/ and submit for a district.. many rows have accuracy and maps link as broken!

### Expected behaviour
Maps co-ordinates should only be comma separated latitude and longitude. 

### Actual behaviour
People can type in anything causing the issue in buttons/display

### Proposed solution
 Revert back to old method of making it read only and only allowing it for self requests (as done in #122 )

**Comment by ajitsen** [comment]
Currently the code is requesting openmaps to do the reverse lookup.
Could  disabling reverse lookup could be the solution ?

```
var success = function(pos) {
			var crd = pos.coords;
				gpsField.value = crd.latitude + "," + crd.longitude;
				gpsAccuracyField.value = crd.accuracy + ' Meters';
				$.get('https://nominatim.openstreetmap.org/reverse?format=jsonv2&lat=' + crd.latitude + '&lon=' + crd.longitude + '&zoom=18&addressdetails=1')
					.done(function(response) {
						var inputElem = $('<input class="form-control" id="id_latlng_display" value="' + response.display_name + '"</input>')
						$(gpsField).before(inputElem);
						$(gpsField).hide();
						console.log(response.display_name);
					})
					.fail(function(err){
						console.log(err);
					});
```

**Comment by irshadshalu** [comment]
> Could disabling reverse lookup could be the solution ?

No, it will only cause more confusion as layman would not be familiar with concept of GPS co-ordinates. The best way is to 

* Implement auto detection when requesting for self.
* Implement Place Picker when requesting for others

#232  disables input if requesting for others which is a temporary fix until a PlacePicker is created.

**Comment by irshadshalu** [comment]
@naveenpf , please do merge #232 .

**Comment by irshadshalu** [comment]
This is temporarily solved as of now, but a placepicker is of urgent requirement.

**Comment by irshadshalu** [comment]
L:
If anyone's working on it, preferred workflow would be

**User clicks on the GPS field -> modal opens -> types address -> co-ordinates are stored.**

(Ability to mark the location on map itself would be a plus)

If you're having API restriction issues, Do look into 

http://wiki.openstreetmap.org/wiki/Nominatim by Openstreetmap.

**Comment by drtechie** [comment]
Not sure if making read only is the ideal solution. 
I recieved a request in WhatsApp with location but I couldn't enter the location coordinates. 

May be a warning to edit this field only if the user knows what he/she is doing will be better?

**Comment by irshadshalu** [comment]
Making it readonly is never intended as a working solution, few people are already working on implementing the PlacePicker.

**Comment by ashwinkjoseph** [comment]
This should be fixed in PR #266

**Comment by msrshahrukh100** [comment]
is anyone working on adding a place picker? - I can add this one - https://developers.google.com/maps/documentation/javascript/examples/places-autocomplete

**Comment by bobinson** [comment]
please check #248 as well.

**Comment by answerquest** [comment]
suggestion for the location picker : Do like ola/uber. Keep a crosshairs/marker fixed, centered over the map. Take the map's center as the lat-long.User only has to move the map around and position it below the marker/crosshairs. Much easier to use on mobiles.

**Comment by bobinson** [comment]
will be fixed with #266

**Comment by irshadshalu** [comment]
Automatic location picking is already done in #29 and is still supported in #289. If possible, test and verify #289 for faster merging.

**Comment by answerquest** [comment]
Hi I've made a simple location picker / placepicker on ola/uber model : move the map, the marker is in the center. I don't think I can build a PR for this as I don't understand the rest of the code and how to push mine in.  
Live: https://answerquest.github.io/leaflet-center-location-picker.html  
Code: https://github.com/answerquest/answerquest.github.io/blob/master/leaflet-center-location-picker.html

**Comment by ashwinkjoseph** [comment]
@answerquest I have made the changes you requested in #266  itself, it behaves like how you described here now, the marker moves with center of the map

**Comment by answerquest** [comment]
@ashwinkjoseph that's great! Thanks!

**Comment by bobinson** [comment]
need to test #266 : please test and report so that this can be merged.
